src/circuit/gadget/poseidon.rs

```diff
@@ -144,7 +145,13 @@ impl<
         chip.initial_state(&mut layouter, &domain)
             .map(|state| Duplex {
                 chip,
-                sponge: Sponge::Absorbing([None; RATE]),
+                sponge: Sponge::Absorbing(
+                    (0..RATE)
+                        .map(|_| None)
+                        .collect::<Vec<_>>()
+                        .try_into()
+                        .unwrap(),
```

Comment:
Why is this change needed?

Reply:
Because of the removal of the `Copy` bound; `let x = [init_val; N];` requires that `init_val` can be copied.